### ISSUE TITLE
captus: specify megahit version number for linux builds

### ISSUE THREAD

**Issue by mptrsen** [bioconda/bioconda-recipes#52888]
Without version number, this can pull in an old version of megahit that is missing critical command line options (such as `--tmp-dir`). As a result, the Captus assembly stage fails, **silently**.

Remove the non-versioned megahit entry in `meta.yaml` and use v1.2.9 for both builds:

https://github.com/bioconda/bioconda-recipes/blob/d378c7122f1cb885fa3c84ceecc062a2391e14e3/recipes/captus/meta.yaml#L32-L33

@edgardomortiz thank you! :pray:

**Comment by edgardomortiz** [comment]
Thanks @mptrsen, maybe you have a suggestion for this specific issue. There is a newer build for OSX than hfbae3c0_0, however the new one has compilation problems and doesn't run properly, that is why I was trying to require this specific older build. Unfortunately the build hfbae3c0_0 is always ignored and one has to install it manually (therefore I added these instructions in our README).

If a specific build can't be invoked I would leave the recipe as:
- megahit =1.2.9

Otherwise I would try:
- megahit =1.2.9 # [linux]
- megahit =1.2.9=hfbae3c0_0 # [osx]

Which would be your suggestion?

Thanks,

Edgardo

**Comment by mptrsen** [comment]
@edgardomortiz thanks for your swift response!

If the OSX build hfbae3c0_0 is ignored and needs to be installed manually anyway then I would simplify the recipe to

```
- megahit =1.2.9
```

Manual installation is suboptimal, I hope the OSX users see the notice in the README.

Thank you!

**Comment by edgardomortiz** [comment]
Yeah, I thought so, I was hoping there was a reliable method to point to a specific build.

Thanks @mptrsen

**Comment by edgardomortiz** [comment]
@mptrsen I opted for the following:
```
    - megahit >=1.2.9 # [linux]
    - megahit =1.2.9=hfbae3c0_0 # [osx]
 ```
The checks passed, just waiting for the merge.

Thanks,

Edgardo

**Comment by mencian** [comment]
This should be solved in #55320.
